libdl/dl_syscalls.h

```diff
@@ -74,22 +74,49 @@
 #define RLIMIT_DLDLINE		16
 #define RLIMIT_DLRTIME		17
 
+/*
+ * For the sched_{set,get}attr() calls
+ */
+#define SCHED_FLAG_RESET_ON_FORK        0x01
+#define SCHED_FLAG_RECLAIM              0x02
+#define SCHED_FLAG_DL_OVERRUN           0x04
+#define SCHED_FLAG_KEEP_POLICY          0x08
+#define SCHED_FLAG_KEEP_PARAMS          0x10
+#define SCHED_FLAG_UTIL_CLAMP_MIN       0x20
+#define SCHED_FLAG_UTIL_CLAMP_MAX       0x40
+
+#define SCHED_FLAG_KEEP_ALL 	(SCHED_FLAG_KEEP_POLICY | \
+				 SCHED_FLAG_KEEP_PARAMS)
+
+#define SCHED_FLAG_UTIL_CLAMP	(SCHED_FLAG_UTIL_CLAMP_MIN | \
+				 SCHED_FLAG_UTIL_CLAMP_MAX)
+
+#define SCHED_FLAG_ALL	(SCHED_FLAG_RESET_ON_FORK	| \
+			 SCHED_FLAG_RECLAIM		| \
+			 SCHED_FLAG_DL_OVERRUN		| \
+			 SCHED_FLAG_KEEP_ALL		| \
+			 SCHED_FLAG_UTIL_CLAMP)
+
 struct sched_attr {
 	__u32 size;
-	
+
 	__u32 sched_policy;
 	__u64 sched_flags;
-	
+
 	/* SCHED_NORMAL, SCHED_BATCH */
 	__s32 sched_nice;
-	
+
 	/* SCHED_FIFO, SCHED_RR */
 	__u32 sched_priority;
-	
+
 	/* SCHED_DEADLINE */
 	__u64 sched_runtime;
 	__u64 sched_deadline;
 	__u64 sched_period;
+
+	/* Utilization hints */
+	__u32 sched_util_min;
+	__u32 sched_util_max;
```

Comment:
I'm sorry I might be too picky here. I thought in the user API of uclamp for cgroup we use uclamp_min and uclamp_max? I think this naming is okay but if we use a different name somewhere else I think for consistency it might be worth to consider using one name across. I don't mind if it stays like this if you're happy with it and no one else comments on it.

Reply:
What you say makes sense, we should keep consistency.
Unfortunately, the cgroup names (`uclamp_{min,max}`) have been requested by Tejun once the core bits were already mainline, hence we (are going to) have now this asymmetry.

Here I'm using the same names used in Linux mainline's `include/uapi/linux/sched.h`.

I would keep rt-app code aligned to whatever we have in kernel-space... although unfortunate.